tools/base/tests/test_utils.py

```diff
@@ -119,39 +120,70 @@ def test_util_coverage_with_data_file(patches):
         == [(m_open.return_value.__enter__.return_value,), {}])
 
 
-def test_util_extract(patches):
+
+@pytest.mark.parametrize(
+    "tarballs",
+    [(), tuple("TARB{i}" for i in range(0, 3))])
+def test_util_extract(patches, tarballs):
     patched = patches(
-        "tempfile.TemporaryDirectory",
+        "nested",
         "tarfile.open",
         prefix="tools.base.utils")
 
-    with patched as (m_tmp, m_open):
-        assert utils.extract("TARBALL", "PATH") == "PATH"
+    with patched as (m_nested, m_open):
+        _extractions = [MagicMock(), MagicMock()]
+        m_nested.return_value.__enter__.return_value = _extractions
+
+        if not tarballs:
+
```

Comment:
Tiny nit: lot of gratuitous blank lines here. Prefer compact spacing.

Reply:
i tend to give `if` and `with` a line above but ill remove the the line below...

Reply:
hmm actually i would prefer to keep - i think with contexts its better to have a clear line above, they are kinda almost like closure functions or something, so i think better to be explicit

Reply:
i reduced it a bit by swapping the conditional around